silhouette/Graphtec.py

```diff
@@ -627,45 +620,31 @@ def initialize(self):
 
     if self.product_id() in [PRODUCT_ID_SILHOUETTE_CAMEO3, PRODUCT_ID_SILHOUETTE_CAMEO4]:
 
-      self.send_command("TB71") # Unknown: 2 five digit numbers. Probably machine stored calibration offset of the regmark sensor optics
-      try:
-        resp = self.read(timeout=1000)
-        if len(resp) > 1:
-          print("TB71: '%s'" % (resp[:-1]), file=self.log)  # response '    0,    0' on portrait
-      except:
-        pass
+      # Unknown: 2 five digit numbers. Probably machine stored calibration offset of the regmark sensor optics
+      resp = self.send_receive_command("TB71")
+      if resp:
+        # response '    0,    0' on portrait
+        print("TB71: '%s'" % resp, file=self.log)
 
-      self.send_command("FA") # Unknown: 2 five digit numbers. Probably machine stored calibration factors of carriage and roller (carriage, roller / unit 1/100% i.e. 0.0001)
-      try:
-        resp = self.read(timeout=1000)
-        if len(resp) > 1:
-          print("FA: '%s'" % (resp[:-1]), file=self.log) # response '0,0'
-                                                      # response '    0,    0' on portrait
-      except:
-        pass
+      # Unknown: 2 five digit numbers. Probably machine stored calibration factors of carriage and roller (carriage, roller / unit 1/100% i.e. 0.0001)
+      resp = self.send_receive_command("FA")
+      if resp:
+        # response '    0,    0' on portrait
+        print("FA: '%s'" % resp, file=self.log)
 
     # Silhouette Studio does not appear to issue this command when using a cameo 4
     if self.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3:
-      self.send_command("TC")
-      try:
-        resp = self.read(timeout=1000)
-        if len(resp) > 1:
-          print("TC: '%s'" % (resp[:-1]), file=self.log) # response '0,0'
-      except:
-        pass
+      resp = self.send_receive_command("TC")
+      if resp:
+        # response '0,0'
+        print("TC: '%s'" % resp, file=self.log)
 
   def get_version(self):
     """Retrieve the firmware version string from the device."""
 
     if self.dev is None: return None
 
-    self.send_command("FG")   # Get Version: 17 Chars
-    try:
-      resp = self.read(timeout=10000) # Large timeout because the plotter moves.
-    except usb.core.USBError as e:
-      print("usb.core.USBError:", e, file=self.log)
-      resp = "None  "
-    return resp[0:-2]   # chop of 0x03
+    return self.send_receive_command("FG", rx_timeout = 10000)
```

Comment:
Here you are chopping one character (0x03) instead of two in the previous implementation. However, I don't see why we would want to remove two tailing characters here. So am I right in assuming, that you didn't saw a reason for removing the last two characters either and therefore changed the behavior accordingly, or am I overseeing something and we should indeed remove two characters?

Reply:
Did you see any response with two times of a 0x03? I had no idea why there was a chop of two chars

Reply:
I could replace it with `.rstrip(CMD_ETX)`

Reply:
No, I did not. That was why I was confused by the old code. I presume it was just a typo or a blackout by the writer, where he or she thought that 0x03 must be two chars '0' and '3', or something. So I guess it makes more sense to chop the one character as also indicated by the comment.

I will mark this as resolved, just wanted to make sure that the change of behavior was intended and not a mistake.